sonic_data_client/db_client.go

```diff
@@ -652,102 +715,121 @@ func tableData2TypedValue(tblPaths []tablePath, op *string) (*gnmipb.TypedValue,
 	return msi2TypedValue(msi)
 }
 
-func enqueFatalMsg(c *DbClient, msg string) {
-	c.q.Put(Value{
+func enqueueFatalMsg(c *DbClient, msg string) {
+	putFatalMsg(c.q, msg)
+}
+
+func putFatalMsg(q *queue.PriorityQueue, msg string) {
+	q.Put(Value{
 		&spb.Value{
 			Timestamp: time.Now().UnixNano(),
 			Fatal:     msg,
 		},
 	})
 }
 
-// for subscribe request with granularity of table field, the value is fetched periodically.
-// Upon value change, it will be put to queue for furhter notification
-func dbFieldMultiSubscribe(gnmiPath *gnmipb.Path, c *DbClient) {
+// dbFieldMultiSubscribe would read a field from multiple tables and put to output queue.
+// For SAMPLE mode, it would send periodically regardless of change.
+// However, if `updateOnly` is true, the payload would include only the changed fields.
+// For ON_CHANGE mode, it would send only if the value has changed since the last update.
```

Comment:
Does line 733,734 imply: In case of STREAM, Sample + update_only(true) is same as onchange?
Also can we pass update_only from client? is there any example I can look at

Reply:
It is a bit different. `ON_CHANGE` pertains to "when" to send data (e.g. after change). However, `updates_only` pertains to "what" to send.
When it is `sample + update_only(true)`, it will still send the update periodically but only includes the updated parts. E.g. empty data in payload if there is no change since the last update.

Currently, only the go client (gnmi_cli) can make this query. I've included an example query and output in the PR description.

For C# client side the usage looks like this:
```
            SubscriptionList sl = new SubscriptionList
            {
                Mode = SubscriptionList.Types.Mode.Stream,
                UseAliases = false,
                Prefix = prefix,
                Encoding = Encoding.JsonIetf,
                UpdatesOnly = options.UpdatesOnly
            };

```